sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -163,6 +163,24 @@ case class DataSourceResolution(
     case DataSourceV2Relation(CatalogTableAsV2(catalogTable), _, _) =>
       UnresolvedCatalogRelation(catalogTable)
 
+    case ShowTablesStatement(None, pattern) =>
+      defaultCatalog match {
+        case Some(_) =>
+          throw new AnalysisException(
+            "v2 catalog doesn't support getCurrentDatabase yet, " +
+              "so the default database name cannot be deduced.")
+        case None =>
+          ShowTablesCommand(None, pattern)
+      }
+
+    case plan @ ShowTablesStatement(Some(namespace), pattern) =>
+      val CatalogObjectIdentifier(maybeCatalog, identifier) = namespace
+      val catalog = maybeCatalog.orElse(defaultCatalog)
```

Comment:
`CatalogTableIdentifier` will fill in the default catalog, right?

Reply:
Yes you are right. `CatalogObjectIdentifier` will fill in the default catalog. I will remove this and also update these above:
```
    case replace: ReplaceTableStatement =>
      // the provider was not a v1 source, convert to a v2 plan
      val CatalogObjectIdentifier(maybeCatalog, identifier) = replace.tableName
      val catalog = maybeCatalog.orElse(defaultCatalog)
```